src/cachemachine/automatedteller.py

```diff
@@ -0,0 +1,75 @@
+import asyncio
+from typing import Any, Dict, List, Sequence, Union
+
+import structlog
+
+from cachemachine.cachechecker import CacheChecker
+from cachemachine.cachedepositer import CacheDepositer
+from cachemachine.rubinrepoman import RubinRepoMan
+from cachemachine.simplerepoman import SimpleRepoMan
+
+logger = structlog.get_logger(__name__)
+
+
+class AutomatedTeller:
+    available_images: List[Dict[str, str]] = []
+    desired_images: List[Dict[str, str]] = []
+    images_to_cache: List[Dict[str, str]] = []
+
+    def __init__(
+        self,
+        name: str,
+        labels: Dict[str, str],
+        repomen: Sequence[Union[SimpleRepoMan, RubinRepoMan]],
+    ):
+        self.name = name
+        self.labels = labels
+        self.repomen = repomen
+
+        self.recommended_image_urls = []
+        for r in self.repomen:
+            if r.recommended_image_url:
+                self.recommended_image_urls.append(r.recommended_image_url)
+
+        self.checker = CacheChecker(self.labels, self.recommended_image_urls)
+        self.depositer = CacheDepositer(self.name, self.labels)
+
+    # Note, doesn't actually return, intended to run forever.
+    async def do_work(self) -> None:
+        while True:
+            self.available_images = []
+            self.desired_images = []
+            self.images_to_cache = []
+
+            self.checker.check()
+
+            if not self.checker.nodes_exist:
+                logger.warning(f"No nodes are labeled with: {self.labels}")
+            else:
+                for r in self.repomen:
+                    recommended_names = self.checker.recommended_names[
+                        r.recommended_image_url
+                    ]
+                    for image in r.desired_images(recommended_names):
+                        self.desired_images.append(image)
```

Comment:
A desired image is an image we want to pull.

recommended_names is the list of other tags that have the same sha hash as recommended.

Reply:
If they all have the same SHA hash, wouldn't we want to pull only one of them, since they're all equivalent?

Reply:
We're not pulling any of them by sha hash, and since it's the sha hash obviously there's only one of them.  But there may be multiple other days, like recommended could be weekly_41.  It could honestly be any number of other tags.  We're passing in the tag names here, not the hashes.

Reply:
Oh, I see my confusion. I thought `desired_images` was selecting some subset of the list of image names it was passed in, but it's not doing anything like that and the names passed in are just used for sorting.